internal/server/singleprocess/service_job.go

```diff
@@ -180,6 +228,136 @@ func (s *service) QueueJob(
 	return &pb.QueueJobResponse{JobId: job.Id}, nil
 }
 
+func (s *service) launchOndemandRunner(
+	ctx context.Context,
+	source *pb.Job,
+	project *pb.Project,
+	odref *pb.Ref_OndemandRunner,
+) (string, error) {
+	log := hclog.FromContext(ctx)
+
+	od, err := s.state.OndemandRunnerGet(odref)
+	if err != nil {
+		return "", err
+	}
+
+	runnerId, err := server.Id()
+	if err != nil {
+		return "", err
+	}
+
+	log.Info("requesting ondemand runner via task start", "runner-id", runnerId)
+
+	// Get our server config
+	cfg, err := s.state.ServerConfigGet()
+	if err != nil {
+		return "", err
+	}
+
+	// Follow the same logic as RunnerGetDeploymentConfig, which includes this note:
+	// Our addr for now is just the first one since we don't support
+	// multiple addresses yet. In the future we will want to support more
+	// advanced choicing.
+
+	var addr *pb.ServerConfig_AdvertiseAddr
+
+	// This should only happen during tests
+	if len(cfg.AdvertiseAddrs) == 0 {
```

Comment:
Our TestServer function should probably initialize the server with this so we don't have to do this check. We can do that another time...